src/ui/marker.js

```diff
@@ -484,4 +510,61 @@ export default class Marker extends Evented {
     isDraggable() {
         return this._draggable;
     }
+
+    /**
+     * Sets the `rotation` property for pointing the marker in a particular direction
+     * @param {number} [rotation=0] Sets the direction the marker is pointing
+     * @returns {Marker} `this`
+     */
+    setRotation(rotation: number) {
+        this._rotation = rotation || 0;
+        this._update();
+        return this;
+    }
+
+    /**
+     * Returns the current rotation angle
+     * @returns {number}
+     */
+    getRotation() {
+        return this._rotation;
+    }
+
+    /**
+     * Sets the `rotateAlignment` property for pointing the marker in a particular direction
+     * @param {string} [alignment='auto'] Sets the reference for marker rotations
+     * @returns {Marker} `this`
+     */
+    setRotationAlignment(alignment: string) {
+        this._rotationAlignment = alignment || 'auto';
+        this._update();
+        return this;
+    }
+
+    /**
+     * Returns the current rotation alignment
+     * @returns {string}
+     */
+    getRotationAlignment() {
+        return this._rotationAlignment;
+    }
+
+    /**
+     * Sets the `pitchAlignment` property for aligning the pitch of a marker with the viewport or the map
+     * @param {string} [alignment='auto'] Sets the pitch alignment property
+     * @returns {Marker} `this`
+     */
+    setPitchAlignment(alignment: string) {
+        this._pitchAlignment = alignment || 'auto';
```

Comment:
This would also need to change to match `rotateAlignment` when `alignment` is `auto`

Reply:
Updated to `alignment || this._rotationAlignment`